src/engine/runtime.js

```diff
@@ -1624,6 +1628,8 @@ class Runtime extends EventEmitter {
      */
     attachStorage (storage) {
         this.storage = storage;
+        fetchWithTimeout.setFetch(storage.scratchFetch.scratchFetch);
```

Comment:
Where is `storage.scratchFetch.scratchFetch` defined and why are we doing this?

Reply:
`scratchFetch` is a new export from `scratch-storage`
- defined here: https://github.com/LLK/scratch-storage/blob/develop/src/scratchFetch.js
- exported here: https://github.com/LLK/scratch-storage/blob/develop/src/ScratchStorage.js#L54-L61

This line tells `fetchWithTimeout` to use `scratchFetch` instead of the global `fetch`. They have the same API, but `scratchFetch` has new features related to metadata headers.

More info about `scratchFetch`:
- LLK/scratch-storage#404
- LLK/scratch-storage#411

Reply:
Ohhh ok yeah that makes sense, thanks for the context!